RecoEgamma/EgammaTools/interface/AnyMVAEstimatorRun2Base.h

```diff
@@ -68,8 +74,13 @@ class AnyMVAEstimatorRun2Base {
   const bool debug_;
 };
 
-// define the factory for this base class
-#include "FWCore/PluginManager/interface/PluginFactory.h"
+// Define the factory for this base class, with forward declared plugin factory
+// such that PluginFactory.h is not included when generating dictionaries for
+// the AnyMVAEstimatorRun2Base on the fly.
+namespace edmplugin {
+  template <class T>
+  class PluginFactory;
+}
 typedef edmplugin::PluginFactory<AnyMVAEstimatorRun2Base*(const edm::ParameterSet&)> AnyMVAEstimatorRun2Factory;
```

Comment:
To me this sounds more of a reason to move the plugin factory definition out from this file.

Reply:
Well, I don't see why that's better, but it's also fine for me :) I have now put it in a separate file called `AnyMVAEstimatorRun2Factory.h`. Are you fine with that?

Reply:
> Well, I don't see why that's better, but it's also fine for me :)

The users of `AnyMVAEstimatorRun2Base` do not need to know about the factory (only the code creating the object needs the factory). This separation is actually a common pattern in CMSSW (I'd even say vast majority of plugin factories are defined that way, but I didn't do a thorough analysis). The split also avoids the need to explicitly forward-declare types from the framework (code rule 4.7).

> I have now put it in a separate file called `AnyMVAEstimatorRun2Factory.h`. Are you fine with that?

That's good, thanks!

Reply:
Ok, got it!